src/core_atmosphere/physics/physics_wrf/module_cu_gf_mpas.F

```diff
@@ -624,36 +682,55 @@ subroutine cu_grell_freitas( &
 
     do i = its, ite
        if(pret(i) .gt. 0. .or. pretm(i).gt.0. .or. prets(i).gt.0.) then
-          xmb_total(i,j) = xmb(i)+xmbm(i)+xmbs(i)
-          pratec(i,j)    = pret(i)+pretm(i)+prets(i)
-          raincv(i,j)    = (pret(i)+pretm(i)+prets(i))*dt
+          vcpool(i,j)      = vcpool_d(i)*frh_out(i)
+          xmb_total(i,j) = factor(i)*(xmb(i)+xmbm(i)+xmbs(i))
+          sig_deep(i,j)=min(.999,1.-frh_out(i))
+          !sig_deep(i,j)=min(.999,frh_out(i))
+          pratec(i,j)    = factor(i)*(pret(i)+pretm(i)+prets(i))
+          raincv(i,j)    = factor(i)*(pret(i)+pretm(i)+prets(i))*dt
+!          if(outt(i,10)*86400. .lt.-1000. .or. subten_t(i,10)*86400. .gt. 1000)print*,'gfmpas1',outt(i,10)*86400.,subten_t(i,10)*86400.
+!          if(outt(i,10)*86400. .lt.-1000. .or. subten_t(i,10)*86400. .gt. 1000)print*,'gfmpas2',outt(i,25)*86400.,subten_t(i,25)*86400.
           ktopmax=max(ktopm(i),ktop(i),ktops(i))
           kbconmax=max(kbconm(i),kbcon(i),kbcons(i))
           
           if(ktopmax > htop(i,j) ) htop(i,j) = ktopmax + .001
           if(kbconmax < hbot(i,j)) hbot(i,j) = kbconmax + .001
+       else if (ierr(i).gt.0) then
+          sig_deep(i,j)=0.
+          factor(i)=0.
        endif
     enddo
 
     !... always save the tendencies of potential temperature, water vapor, cloud water, and cloud ice:
-    do k = kts, ktf
-       do i = its, itf
-          rthcuten(i,k,j) = (outts(i,k) + outt(i,k)+outtm(i,k))/pi(i,k,j)
-          rqvcuten(i,k,j) = outqs(i,k) + outq(i,k)+outqm(i,k)
-          rucuten(i,k,j) = outu(i,k)+outum(i,k)+outus(i,k)
-          rvcuten(i,k,j) = outv(i,k)+outvm(i,k)+outvs(i,k)
-          gf_mfx(i,k) = xmb(i)*zuo(i,k)/(1.e-8+forcing(i,6))+xmbm(i)*zum(i,k)+xmbs(i)*zus(i,k)
-
+    do k = kts, kte
+       do i = its, ite
+          rthcuten(i,k,j) = factor(i)*(outts(i,k) + outt(i,k)+outtm(i,k))/pi(i,k,j)
+          rqvcuten(i,k,j) = factor(i)*(outqs(i,k) + outq(i,k)+outqm(i,k))
+          rucuten(i,k,j) = factor(i)*(outu(i,k)+outum(i,k)+outus(i,k))
+          rvcuten(i,k,j) = factor(i)*(outv(i,k)+outvm(i,k)+outvs(i,k))
+          sub3d_rthcuten(i,k,j) = factor(i)*subten_t(i,k)/pi(i,k,j)
+          sub3d_rqvcuten(i,k,j) = factor(i)*subten_q (i,k)
+          sub3d_rucuten (i,k,j) = factor(i)*subten_u (i,k)
+          sub3d_rvcuten (i,k,j) = factor(i)*subten_v (i,k)
+
+!---temporary setting -- must be removed once the subsidence tendencies are correct
+                  !sub3d_rthcuten(i,k,j) = 2.e-5
```

Comment:
Are they correct yet?

Reply:
Yes, but this may be useful for later tests ?